datafusion/sql/src/unparser/expr.rs

```diff
@@ -958,8 +958,14 @@ impl Unparser<'_> {
             }
             DataType::Float32 => Ok(ast::DataType::Float(None)),
             DataType::Float64 => Ok(ast::DataType::Double),
-            DataType::Timestamp(_, _) => {
-                not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
+            DataType::Timestamp(_, tz) => {
+
+                let tz_info = match tz {
+                    Some(_) => TimezoneInfo::Tz,
```

Comment:
Tz is specific to the Postgres dialect: https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/ast/data_type.rs#L632

To make this compatible with more systems, we should use `WithTimeZone`

Reply:
Updated, thank you